core/storage/collection/gae_models.py

```diff
@@ -102,7 +114,12 @@ def get_instance_id(cls, collection_id, version):
 
     @classmethod
     def get_all_non_private_commits(
-            cls, page_size, urlsafe_start_cursor, max_age=None):
+            cls,
+            page_size, # type: int
+            urlsafe_start_cursor, # type: Optional[Text]
+            max_age=None # type: Optional[datetime.timedelta]
+    ):
+        # type: (...) -> Tuple[List[CollectionCommitLogEntryModel], Optional[Text], bool]
```

Comment:
A function that returns a tuple of three values where first is a list of some type of models, second is a cursor, and the third is a boolean is in a few places in the codebase. Would it be possible to make a generic type variable for it?

Reply:
Hi @vojtechjelinek -- I think, if we do that, it should probably be a generic SearchResult domain object. Wdyt?

Reply:
It is possible to make a generic type variable for it. 
@seanlip Do you think using domain object will be a nice thing to do? I will add this in the issue created for this. Please let me know. Thanks!!

Reply:
Yes, I think so -- if you have stuff that's passed around internally to the Oppia codebase, then in general, I would suggest a domain object. Thanks for checking!

Reply:
Have added it to the issue. Thanks!!

Reply:
Yeah, domain object makes more sense here, I think mainly because we have a chain of methods that return 3-tuple. If there was just one function in that change custom type might be better